File: rust/src/models/components_scope.rs
// this file is @generated
use serde::{Deserialize, Serialize};

/// Component names — matched against `ReplacePlanRequest::components[].name`.
#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct ComponentsScope {
    pub component_names: Vec<String>,
}

impl ComponentsScope {
    pub fn new(component_names: Vec<String>) -> Self {
        Self { component_names }
    }
}

File: typescript/src/models/componentsScope.ts
// this file is @generated
/** Component names — matched against `ReplacePlanRequest::components[].name`. */
export interface ComponentsScope {
  componentNames: string[];
}

export const ComponentsScopeSerializer = {
  _fromJsonObject(object: any): ComponentsScope {
    return {
      componentNames: object["component_names"],
    };
  },

  _toJsonObject(self: ComponentsScope): any {
    return {
      component_names: self.componentNames,
    };
  },
};
